lib/active_merchant/billing/gateways/elavon.rb

```diff
@@ -152,216 +184,196 @@ def supports_scrubbing?
+        xml.ssl_cvv2cvc2_indicator  1
       end
 
-      def add_customer_data(form, options)
-        form[:email] = truncate(options[:email], 100) unless empty?(options[:email])
-        form[:customer_code] = truncate(options[:customer], 10) unless empty?(options[:customer])
-        form[:customer_number] = options[:customer_number] unless empty?(options[:customer_number])
-        options[:custom_fields]&.each do |key, value|
-          form[key.to_s] = value
-        end
+      def add_customer_email(xml, options)
+        xml.ssl_email truncate(options[:email], 100) unless empty?(options[:email])
       end
 
-      def add_salestax(form, options)
-        form[:salestax] = options[:tax] if options[:tax].present?
+      def add_salestax(xml, options)
+        return unless options[:tax].present?
+
+        xml.ssl_salestax options[:tax]
       end
 
-      def add_address(form, options)
+      def add_address(xml, options)
         billing_address = options[:billing_address] || options[:address]
 
         if billing_address
-          form[:avs_address]    = truncate(billing_address[:address1], 30)
-          form[:address2]       = truncate(billing_address[:address2], 30)
-          form[:avs_zip]        = truncate(billing_address[:zip].to_s.gsub(/[^a-zA-Z0-9]/, ''), 9)
-          form[:city]           = truncate(billing_address[:city], 30)
-          form[:state]          = truncate(billing_address[:state], 10)
-          form[:company]        = truncate(billing_address[:company], 50)
-          form[:phone]          = truncate(billing_address[:phone], 20)
-          form[:country]        = truncate(billing_address[:country], 50)
+          xml.ssl_avs_address     truncate(billing_address[:address1], 30)
+          xml.ssl_address2        truncate(billing_address[:address2], 30)
+          xml.ssl_avs_zip         truncate(billing_address[:zip].to_s.gsub(/[^a-zA-Z0-9]/, ''), 9)
+          xml.ssl_city            truncate(billing_address[:city], 30)
+          xml.ssl_state           truncate(billing_address[:state], 10)
+          xml.ssl_company         truncate(billing_address[:company], 50)
+          xml.ssl_phone           truncate(billing_address[:phone], 20)
+          xml.ssl_country         truncate(billing_address[:country], 50)
         end
 
         if shipping_address = options[:shipping_address]
-          first_name, last_name = split_names(shipping_address[:name])
-          form[:ship_to_first_name]     = truncate(first_name, 20)
-          form[:ship_to_last_name]      = truncate(last_name, 30)
-          form[:ship_to_address1]       = truncate(shipping_address[:address1], 30)
-          form[:ship_to_address2]       = truncate(shipping_address[:address2], 30)
-          form[:ship_to_city]           = truncate(shipping_address[:city], 30)
-          form[:ship_to_state]          = truncate(shipping_address[:state], 10)
-          form[:ship_to_company]        = truncate(shipping_address[:company], 50)
```

Comment:
Similar to the above question: Are these fields not supported? [I see them listed here](https://developer.elavon.com/#/api/eb6e9106-0172-4305-bc5a-b3ebe832f823.rcosoomi/documents?converge-integration-guide/book/api_reference/../../book/api_reference/supported_transaction_input_fields.html), but I'm not sure what that means.